src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,210 @@
+            if (index == 0) {
+                newNode.setNext(head);
+                newNode.setPrev(null);
+                head.setPrev(newNode);
+                head = newNode;
+                size++;
+            } else if (index == size) {
+                newNode.setNext(null);
+                newNode.setPrev(tail);
+                tail.setNext(newNode);
+                tail = newNode;
+                size++;
+            } else {
+                checkCorrectIndexAdd(index);
+                Node<T> currentNode = searchPosition(index);
+                newNode.setNext(currentNode);
+                newNode.setPrev(currentNode.getPrev());
+                currentNode.getPrev().setNext(newNode);
+                currentNode.setPrev(newNode);
+                size++;
+                if (head == currentNode) {
+                    head = newNode;
+                }
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkCorrectIndexGetRemove(index);
+        Node<T> currentNode = searchPosition(index);
+        return currentNode.getValue();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkCorrectIndexGetRemove(index);
+        Node<T> currentNode = searchPosition(index);
+        T previousValue = currentNode.getValue();
+        currentNode.setValue(value);
+        return previousValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkCorrectIndexGetRemove(index);
+        Node<T> currentNode = searchPosition(index);
+        unlink(currentNode);
+        return currentNode.getValue();
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (currentNode.getValue() == object || (currentNode.getValue() != null
+                    && currentNode.getValue().equals(object))) {
+                unlink(currentNode);
+                return true;
+            }
+            currentNode = currentNode.getNext();
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addFirstNode(T value, Node<T> newNode) {
+        head = newNode;
+        tail = newNode;
+        size++;
+    }
+
+    private void checkCorrectIndexAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index" + index);
+        }
+    }
+
+    private void checkCorrectIndexGetRemove(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Wrong index" + index);
+        }
+    }
```

Comment:
you use it only in `add` method, it would be better to add `if (index == size)` before `checkIndex`